engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -140,41 +175,45 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
       stay
   }
 
+  private def recordHashError(reason: Throwable) = {
+    log.error("{}: hash error: {}. Disabling call caching for this job", tag, reason.getMessage)
+    effectiveCallCachingMode = CallCachingOff
+  }
+
   def prepareJob() = {
-    val jobPreparationActorName = s"$workflowId-BackendPreparationActor-${jobKey.tag}"
+    val jobPreparationActorName = s"BackendPreparationActor_for_$jobTag"
     val jobPrepProps = JobPreparationActor.props(executionData, jobKey, factory, initializationData, serviceRegistryActor)
     val jobPreparationActor = context.actorOf(jobPrepProps, jobPreparationActorName)
     jobPreparationActor ! JobPreparationActor.Start
     goto(PreparingJob)
   }
 
   def initializeJobHashing(jobDescriptor: BackendJobDescriptor, activity: CallCachingActivity) = {
-    val fileHasherActor = context.actorOf(FileHasherActor.props)
-    context.actorOf(EngineJobHashingActor.props(jobDescriptor, fileHasherActor, activity))
+    context.actorOf(EngineJobHashingActor.props(jobDescriptor, initializationData, factory.fileContentsHasherActor, dockerHashLookupActor, factory.runtimeAttributeDefinitions, backendName, activity), s"ejha_for_$jobDescriptor")
```

Comment:
Is it reasonable to just pass the factory to avoid passing two arguments that here derive from the factory, or does that not generalize?